src/app/modules/user/user.service.ts

```diff
@@ -118,18 +119,31 @@ export const verifyContactOtp = async (
 }
 
 /**
- * Updates the user document with the userId with the given contact.
+ * Updates the user document with the userId with the given contact and returns
+ * the populated updated user.
  * @param contact the contact to update
  * @param userId the user id of the user document to update
+ * @returns the updated user with populated references
+ * @throws error if any db actions fail
  */
 export const updateUserContact = async (contact: string, userId: string) => {
   // Retrieve user from database.
   // Update user's contact details.
-  const admin = await User.findById(userId)
+  const admin = await User.findById(userId).populate({
+    path: 'agency',
+    model: AGENCY_SCHEMA_ID,
+  })
   if (!admin) {
     throw new Error('User id is invalid')
   }
 
   admin.contact = contact
   return admin.save()
 }
+
+export const getPopulatedUserById = async (userId: string) => {
```

Comment:
Might inline this into schema, but I don't want controllers to call the DB layer, so will probably still need a UserService function. Comments?